libraries/AP_HAL_Linux/UARTDriver.cpp

```diff
@@ -429,15 +346,7 @@ int UARTDriver::_write_fd(const uint8_t *buf, uint16_t n)
  */
 int UARTDriver::_read_fd(uint8_t *buf, uint16_t n)
 {
-    int ret;
-
-    ret = _device->read(buf, n);
-
-    if (ret > 0) {
-        BUF_ADVANCETAIL(_readbuf, ret);
```

Comment:
this is correct, but see the comment on SPIUARTDriver... it's probably wrong there.